common/sky/moon.py

```diff
@@ -0,0 +1,481 @@
+import numpy as np
+from astropy.time import Time
+from astropy.coordinates import SkyCoord, EarthLocation, Distance, GeocentricTrueEcliptic, Angle
+import astropy.units as u
+from astropy.time import TimeDelta
+from common.sky.utils import local_sidereal_time, current_geocent_frame, geocentric_coors, hour_angle_to_angle
+from common.sky.constants import J2000, EQUAT_RAD
+from common.sky.altitude import Altitude
+from typing import NoReturn, Tuple
+
+
+class Moon:
+
+    def __init__(self):
+
+        #TODO: skip constructor altogether and just use the at method?
+        self.time = None
+        self.time_jd = None
+        self.time_ttjd = None  # important to use correct time argument for this!!
+        self.scalar_input = False
+        self.PIE = None
+        self.LAMBDA = None
+        self.BETA = None
+
+    def at(self, time: Time):
+
+        self.time = time
+        self.time_jd = np.asarray(time.jd)
+        self.time_ttjd = np.asarray(time.tt.jd)  # important to use correct time argument for this!!
+        self.scalar_input = False
+        if self.time_ttjd.ndim == 0:
+            self.time_ttjd = self.time_ttjd[None]  # Makes 1D
+            self.scalar_input = True
+        
+        return self
+
+    def _low_precision_calculations(self) -> NoReturn:
+        """
+        Compute low precision values for the moon location method calculations.
+        """
+        #jd = np.asarray(self.time.jd)
+        T = (self.time_jd - J2000) / 36525.  # jul cent. since J2000.0
```

Comment:
Something feels just little bit dirty to me about having this method use the member variable `self.time_jd` which isn't fixed (otherwise this would be a static calculation) and `self.time_jd` being set in the `at` method. I think, thought, that I'm going to have to look at this up close in my repo to really browse through it since it's a substantial change, so let's focus on getting it merged in.

I'm just worried that we'll end up doing a lot more work (and being a lot less functional, when I am aiming to make this as functional as we can in Python).

When you merge it, please don't delete this PR / branch because I really want to refer to this  PR on my machine.